### DISCUSSION TITLE
Bucket usage not updating

### DISCUSSION THREAD

**Discussion by BlackWand1** [minio/minio#18279] [Q&A] [answered]
Hello,
I have a bucket (maybe more) where usage wont update. I am using prometheus for metrics gathering and this one bucket that I know of, wont update. The value has been 1157 for a very long time even though data is uploaded constantly.

Other buckets increases continuasly.
I have tried restarting cluster.
I have also set scanner speed to FAST and still no change.

minio_bucket_usage_total_bytes{bucket="bucket-1001729-01", instance="***:9000", job="c1p1-job-bucket", server="MINIO-C01P01-01:9000"} | 1157

Running version: 2023-09-04 on Red Hat ppcle64

**Comment by BlackWand1** [comment]
Just upgraded to 2023-10-16 version. Will have to wait at least 12 hours I suppose before I would be able to see any changes.
While the scanner is running, shouldn't the bucket usage increase during the scan, or will it jump up to the new total value once scanner is done?

And even if value is incorrect, shouldn't it be increased with the items that gets uploaded right now?

**Reply by klauspost** [reply, marked as answer]
> While the scanner is running, shouldn't the bucket usage increase during the scan, or will it jump up to the new total value once scanner is done?

When it can, it will do continuous updates - that will be for folders it hasn't seen before. For folders that existed at the last scan that is not possible and total will only be updated when the full folder has been scanned.

You can follow progress with `mc admin scanner info ALIAS`.

**Reply by BlackWand1** [reply]
Of course ;)

I cant get my mc admin to work. I have added "local" in the json file with access & secret to an "account".
All I get is this: ./mc admin info local
mc: <ERROR> Unable to get service status. Access Denied.

Maybe it has to be an account with extra privileges?

**Reply by BlackWand1** [reply]
Created new account with consoleAdmin and new keys. Now mc admin works.
And I can check the scanner info. Thanks.

**Reply by Clara12062** [reply]
@klauspost  I had the same problem. 
Can I think of it this way: as the number of objects in the bucket increases, the scan time increases, and the bucket usage updates become slower. 
When there are a lot of objects in the bucket, even if I adjust the duration of the cycle, it doesn't have much effect, right?

**Reply by klauspost** [reply]
The cycle parameter is deprecated and does not affect scan speed. Use the speed parameter.

Scan speed is a function of number of objects and reduced by number of drives. More objects = slower updates. More disks = faster scanning.

**Reply by Clara12062** [reply]
@klauspost  Thanks for your reply!

I have another question.
I don't have any extra disks, and I plan to use multiple buckets instead, each bucket being a directory on the disk. 
Does the scanning speed increase in this scenario?
